### PR TITLE
DROOLS-2302: [DMN Designer] Outline is not refreshed after new editor type selected

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2302

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGridGeneralTest.java [hunk 1]
```diff
@@ -145,6 +150,45 @@ public void testGetLayerGridNotAttachedToLayer() {
                      grid.getLayer());
     }
 
+    @Test
+    public void testSelectFirstCellWithNowRowsOrColumns() {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGridGeneralTest.java]: Typo? testSelectFirstCellWithNowRo**w**sOrColumns

Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGridGeneralTest.java]: Typo? testSelectFirstCellWithNo**W**RowsOrColumns

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/function/FunctionGrid.java [hunk 2]
```diff
@@ -290,17 +289,9 @@ public void setExpressionType(final ExpressionType type) {
...
-            @Override
-            public void execute() {
-                gridLayer.draw();
-                oEditor.ifPresent(gridLayer::select);
-            }
-        });
+    @Override
+    protected void synchroniseViewWhenExpressionEditorChanged(final Optional<BaseExpressionGrid> oEditor) {
+        super.synchroniseViewWhenExpressionEditorChanged(oEditor);
         getEditorControls();
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/function/FunctionGrid.java]: Just question, I checked other implementations (DecisionTable, Relation etc.) of the `BaseExpressionGrid`, they do not override `synchroniseViewWhenExpressionEditorChanged` but they correctly show controls if the editor definition changed, so just interested if we really have to override `synchroniseViewWhenExpressionEditorChanged` in `FunctionGrid`

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGridGeneralTest.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGridGeneralTest.java]: Well spotted. Fixing.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/function/FunctionGrid.java (hunk 2 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/function/FunctionGrid.java]: ```FunctionGrid``` sets up different "controls" depending on the selected ```KIND``` and optional Expression Type so needs to also call ```getEditorControls()```. No other Expression Editor needs to perform such a task. In essence the ```FunctionGrid``` is three editors wrapped into one: FEEL, JAVA and PMML each of which has _slightly_ different controls.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**